frame/dvm/ethereum/src/lib.rs

```diff
@@ -934,11 +958,19 @@ impl<T: Config> InternalTransactHandler for Pallet<T> {
 		})?
 	}
 
+	/// Execute transaction from other pallets(internal transaction)
+	/// NOTE: The difference between the rpc transaction and the internal transaction is that
+	/// The internal transactions will catch and throw evm error comes from runner to caller.
+	fn internal_transact(target: H160, input: Vec<u8>) -> DispatchResultWithPostInfo {
+		let source = T::PalletId::get().derive_ethereum_address();
+		Self::internal_transact_with_source_account(source, target, input)
+	}
+
 	/// Pure read-only call to contract, the sender is pallet dvm account.
 	/// NOTE: You should never use raw call for any non-read-only operation, be carefully.
 	fn read_only_call(contract: H160, input: Vec<u8>) -> Result<Vec<u8>, DispatchError> {
 		sp_io::storage::start_transaction();
-		let source = T::PalletId::get().into_h160();
+		let source = T::PalletId::get().derive_ethereum_address();
```

Comment:
Same with above and should move source to parameter if wormhole/helix pallet and handler is removed.